src/Rules/PHPUnit/AssertEqualsIsDiscouragedRule.php

```diff
@@ -0,0 +1,64 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\PHPUnit;
+
+use PhpParser\Node;
+use PhpParser\NodeAbstract;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\GeneralizePrecision;
+use PHPStan\Type\TypeCombinator;
+use function count;
+use function strtolower;
+
+/**
+ * @implements Rule<NodeAbstract>
+ */
+class AssertEqualsIsDiscouragedRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return NodeAbstract::class;
```

Comment:
I feel like they have a more specific class parent than just NodeAbstract. Feel free to check out and optimize other similar rules as well.

Reply:
fixed here and will do for more rules in a separate PR with a 1.4.x target in https://github.com/phpstan/phpstan-phpunit/pull/217